app/src/org/commcare/tasks/DataPullTask.java

```diff
@@ -156,7 +156,7 @@ protected void onCancelled() {
             //This should be per _user_, not per app
             prefs.edit().putLong("last-ota-restore", new Date().getTime()).commit();
 
-            HttpRequestGenerator requestor = new HttpRequestGenerator(username, password);
+            HttpRequestEndpoints requestor = dataPullRequester.getHttpGenerator(username, password);

```

Comment:
route http generator creation through `DataPullResponseFactory` so we can easily override it in tests.